segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/ChunkMetadataStore.java

```diff
@@ -78,107 +81,116 @@
     /**
      * Begins a new transaction.
      *
+     * @param keysToLock Array of keys to lock for this transaction.
      * @return Returns a new instance of {@link MetadataTransaction}.
-     * @throws StorageMetadataException Exception related to storage metadata operations.
      */
-    MetadataTransaction beginTransaction() throws StorageMetadataException;
+    MetadataTransaction beginTransaction(String... keysToLock);
 
     /**
      * Retrieves the metadata for given key.
      *
      * @param txn Transaction.
      * @param key key to use to retrieve metadata.
      * @return Metadata for given key. Null if key was not found.
-     * @throws StorageMetadataException Exception related to storage metadata operations.
+     * @throws CompletionException If the operation failed, it will be completed with the appropriate exception. Notable Exceptions:
+     * {@link StorageMetadataException} Exception related to storage metadata operations.
      */
-    StorageMetadata get(MetadataTransaction txn, String key) throws StorageMetadataException;
+    CompletableFuture<StorageMetadata> get(MetadataTransaction txn, String key);
```

Comment:
Update Javadoc to reflect that you're returning a future.

Reply:
fixed